### DISCUSSION TITLE
xray + new BurpSuite

### DISCUSSION THREAD

**Discussion by nonrect** [chaitin/xray#1767] [Q&A]
hey

I have xRay + latest BurpSuite Pro v2023.10.3.6

I want to use BurpSuite crawler on xray, the guide on "https://docs.xray.cool" is for older versions of burpsuite

in the new version I found "Upstream proxy servers" on "Settings > Connections" and added a value like this:
  • Destination host:  *
  • Proxy host:  127.0.0.1
  • Proxy port: 1119     (something random and open)

now on xray I give this command:
  • ./xray_windows_amd64.exe webscan --listen 127.0.0.1:1119 --html-output "Sample_output.html" --plugins sqldet,cmd-injection,upload

now that xray is in listening mode I go to BurpSuite  >  New scan  >  Crawl (only)  >  set my url in "URLs to scan"  >  OK

There are two senarios, 1 It start crawling and adds only the base url on xray, two it gives an error on burp side by a pause indicating it can't crawl the target url (target url is UP and running well)

What am I missing ?

**Comment by Henrize** [comment]
We tested the issue you mentioned using BurpSuite Pro v2024.6 and xray 1.9.11:

First, we launched xray with the command you provided and configured BurpSuite's Upstream proxy servers to 127.0.0.1:1119.
Afterward, from the Dashboard, we initiated a New scan, selected Crawl (only), added `https://xray.cool/` to the URLs to scan, confirmed with the Lightweight scan mode.

Upon completion of the scan, we observed that xray scanned addresses such as `https://cdn.jsdelivr.net/` and `https://stats.g.doubleclick.net/`, and there were no error outputs reported during the crawling process in BurpSuite.

<img width="1280" alt="image" src="https://github.com/chaitin/xray/assets/34267732/a38a7c93-5085-46e1-aaf6-3e0bed2ffb02">

<img width="1088" alt="image" src="https://github.com/chaitin/xray/assets/34267732/2282a6d2-bbf8-4683-adc4-d670bcaddb21">

If there indeed exists an issue with the collaboration between xray and BurpSuite's Crawler, or if we have misunderstood your concern. Please provide additional information to help us reproduce and address the issue. 

Thank you!
